### PR TITLE
Feature - Update Tag

### LINKED ISSUE #116: Feature - Update Tag command
This task will cover the following:
- [ ] Updating the existing Tag store to support updating
- [ ] Editing the data file to reflect the changes made
- [ ] Adding command for Tag editing

### PR DESCRIPTION
**Pull request by KwanHW**
# Changes
This PR adds the command to allow the user to update a specified tag.

# Related Issues
- Closes #116

### CODE REVIEW

**Review by sharpstorm — CHANGES_REQUESTED** [review]
I noticed only minor issues, overall code and logic looks good.

File: src/main/java/seedu/contax/model/AddressBook.java [hunk 1]
```diff
@@ -176,10 +176,14 @@ private void setPersonsWithTag(Tag target, Tag editedTag) {
...
         requireNonNull(editedTag);
 
-        List<Person> personsWithTag = getPersonList().filtered(person -> person.hasTag(target));
+        List<Person> persons = getPersonList();
+        for (int i = 0; i < persons.size(); i++) {
+            Person person = persons.get(i);
+
+            if (!person.hasTag(target)) {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply src/main/java/seedu/contax/model/AddressBook.java:185]: I found that `filtered` causes the loop to terminate early and hence does not update all tags. Not sure if its an issue with `FilteredList`.
Suppose Person A and Person B are the only persons who need to have their tags updated.
Using the old implementation only Person A will have the updated tag.

[the remark this answers is not in the document]
Comment (sharpstorm) [reply src/main/java/seedu/contax/model/AddressBook.java:185]: Oh this is a different implementation from my streams, that is probably why. Oaky then, this looks fine.

File: src/main/java/seedu/contax/logic/parser/EditTagCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditTagCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap argumentMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NEWTAGNAME);
+
+        // Unlike the other edit commands, Tag only contains the tag name field. Hence, if the new tag name and/or
+        // index is not present,the command must be wrong
+        if (!arePrefixesPresent(argumentMultimap, PREFIX_NEWTAGNAME) || argumentMultimap.getPreamble().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/parser/EditTagCommandParser.java]: Resolved in 34fca78d.

File: src/main/java/seedu/contax/logic/parser/CliSyntax.java [hunk 3]
```diff
@@ -12,6 +12,9 @@
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
     public static final Prefix PREFIX_TAG = new Prefix("t/");
 
+    /* Prefix definition for tag management */
+    public static final Prefix PREFIX_NEWTAGNAME = new Prefix("t/");
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/parser/CliSyntax.java]: Resolved in 34fca78d.

File: src/main/java/seedu/contax/logic/commands/EditTagCommand.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            throw new CommandException(Messages.MESSAGE_INVALID_TAG_DISPLAYED_INDEX);
+        }
+
+        Tag tagToEdit = tagList.get(index.getZeroBased());
+        Tag editedTag = createEditedTag(tagToEdit, editTagDescriptor);
+
+        // Checks if the specified tag name already exists in the address book.
+        if (model.hasTag(editedTag)) {
+            throw new CommandException(MESSAGE_TAG_EXISTS);
+        }
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/commands/EditTagCommand.java:75]: Resolved test wrong test case name in 34fca78d.

File: src/main/java/seedu/contax/logic/commands/EditTagCommand.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+/**
+ * Edits the details of an existing tag in the address book.
+ */
+public class EditTagCommand extends Command {
+    public static final String COMMAND_WORD = "edittag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the specified tag.\n"
+            + "Parameters: INDEX t/NEW_TAGNAME\n"
+            + "Example: " + COMMAND_WORD + " INDEX " + "t/Prospective Clients";
```
[the remark this answers is not in the document]
Comment (KwanHW) [reply src/main/java/seedu/contax/logic/commands/EditTagCommand.java]: Resolved in 34fca78d.

**Review by sharpstorm — APPROVED** [review]
LGTM.

**Review by dandaandaaaaaan — APPROVED** [review]
lgtm!

**Pull request merged**